src/api/base/index.ts

```diff
@@ -0,0 +1,92 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { SupportedChainId } from '../../constants/chains'
+import { CowError } from '../../utils/common'
+import { Context } from '../../utils/context'
+import { Options } from '../cow/types'
+
+interface ConstructorParams {
+  context: Context
+  name: string
+  apiVersion?: string
+  // we want getApiUrl to accept any parameters but return a specific type
+  getApiUrl: (...params: any[]) => Partial<Record<SupportedChainId, string>>
```

Comment:
I don't know much about the best practice here, but shouldn't it instead be `unknown` rather than `any`?

Reply:
it's a pure function